.changeset/popular-snakes-report.md

```diff
@@ -0,0 +1,5 @@
+---
+"@cloudflare/workers-shared": patch
+---
+
+chore: Rename asset-worker RPC methods as unstable\_\*
```

Comment:
what is the purpose we are doing this for? what does `unstable` mean in this context? does it express that these APIs are subject to change?

Reply:
Because we have set up the Asset Worker as an EntryPoint class it is able to accept RPC calls to any of its public methods. Greg doesn't want developers in their User Worker to call these methods directly and become part of our public API contract, so he is naming them with the `unstable_` prefix to mark them as not part of that contradt.

I agree with the sentiment although the correct approach is for these not to be on the public API in the first place, which we could achieve by having a separate named entry-point to this Worker that only the Router Worker gets access to.

In the meantime this is a hack to avoid the public API contract, which I am happy to go with, but with the intention of a fast follow up later this autumn.

Reply:
yeah, I figured as much. Do we need to explicitly call that out maybe? As a comment or co?

Reply:
in fact, lemme do that real quick so we can merge this in

Reply:
ok, done ✅